Code/Framework/AzFramework/AzFramework/Archive/ZipDirCacheFactory.cpp

```diff
@@ -606,142 +593,77 @@ namespace AZ::IO::ZipDir
+                    || pFileHeader->nLastModTime != localFileHeader->nLastModTime)
+                {
+                    AZ_Warning("Archive", false, "ZD_ERROR_VALIDATION_FAILED:"
+                        " The local file header's modification timestamps don't match that of the global file header in the archive."
+                        " The archive timestamps are inconsistent and may be damaged. Check the archive file.");
+                    // don't return here, it may be ok.
+                }
 
-    //////////////////////////////////////////////////////////////////////////
-    // reads the file pointed by the given header and entry (they must be coherent)
-    // and decompresses it; then calculates and validates its CRC32
-    void CacheFactory::Validate(const FileEntryBase& fileEntry)
-    {
-        AZStd::vector<char> pBuffer;
-        // validate the file contents
-        // allocate memory for both the compressed data and uncompressed data
-        pBuffer.resize(fileEntry.desc.lSizeCompressed + fileEntry.desc.lSizeUncompressed);
-        char* pUncompressed = &pBuffer[fileEntry.desc.lSizeCompressed];
-        char* pCompressed = &pBuffer[0];
+                // Validate data
+                if (pFileHeader->desc != localFileHeader->desc  // this checks CRCs and compressed/uncompressed sizes
+                    || pFileHeader->nMethod != localFileHeader->nMethod
+                    || pFileHeader->nFileNameLength != localFileHeader->nFileNameLength)
+                {
+                    AZ_Warning("Archive", false, "ZD_ERROR_VALIDATION_FAILED:"
+                        " The local file header descriptor doesn't match basic parameters declared in the global file header in the file."
+                        " The archive content is inconsistent and may be damaged. Please try to repair the archive.");
+                    // return here because further checks aren't worse than this.
+                    return;
+                }
 
-        AZ_Assert(fileEntry.nFileDataOffset != FileEntry::INVALID_DATA_OFFSET, "File entry has invalid data offset of %" PRIx32, FileEntry::INVALID_DATA_OFFSET);
-        Seek(fileEntry.nFileDataOffset);
+                // Read extra data
+                uint32_t extraDataLen = localFileHeader->nFileNameLength + localFileHeader->nExtraFieldLength;
+                buffer.resize_no_construct(buffer.size() + extraDataLen);
+                Read(buffer.data() + buffer.size(), extraDataLen);
 
-        Read(pCompressed, fileEntry.desc.lSizeCompressed);
+                // Compare local file name with the CDR file name, they should match
+                auto zipFileNameBegin = buffer.data() + sizeof(ZipFile::LocalFileHeader);
+                auto zipFileNameEnd = zipFileNameBegin + localFileHeader->nFileNameLength;
+                if (!AZStd::equal(zipFileNameBegin, zipFileNameEnd, reinterpret_cast<const char*>(pFileHeader + 1)))
```

Comment:
We can make a string_view out of the local file header filename and the CDR filename.
This would help with debugging as well.

```suggestion
                AZStd::string_view zipFileName{ buffer.data() + sizeof(ZipFile::LocalFileHeader), localFileHeader->nFileNameLength};
                AZStd::string cdrZipFileName{ reinterpret_cast<const char*>(pFileHeader + 1), pFileHeader->nFileNameLength
                if (zipFileName != cdrZipFileName))
```

Reply:
Nice, I like it!